Documentation-only: Add Windows specifics for `process_status` type

shym opened the pull request:
Add a paragraph to `unix.mli` and `unixLabels.mli` about the `process_status` type to:
- clarify that only `WEXITED` is ever used on Windows (no signal there),
- and refer to `NTSTATUS` documentation for the possible return codes to indicate errors.
I think `NTSTATUS` gives the proper keyword to find the full [list of codes] I was missing to decode errors on Windows.

[list of codes]: https://learn.microsoft.com/en-us/openspecs/windows_protocols/ms-erref/596a1078-e883-4972-9bbc-49e60bebca55

shym committed:
documentation: Add Windows specifics for process_status type

Clarify that only WEXITED is ever used on Windows
Refer to NTSTATUS documentation for the possible return codes to
indicate errors
Detail the special case of a segfault, quite possibly the most common
case users would encounter

dra27 commented:
I wonderful if it’s worth including a link to https://learn.microsoft.com/en-us/openspecs/windows_protocols/ms-erref/596a1078-e883-4972-9bbc-49e60bebca55 (I don’t know how stable that would be) or perhaps noting that where in Unix you get an uncaught SIGSEGV, on Windows you get exit code 0xC0000005 (-1,073,741,818). (I’ve not double-checked those numbers!)

shym committed:
Add Changes entry for Windows Unix.process_status details

shym commented:
I indeed assumed that a direct URL to the Windows documentation would be fragile. But the case of Segfault is maybe the most commonly encountered one, so I updated the PR with the details of that particular case.
I also updated Changes, as a separate commit to easily drop it if need be (I’m not sure this is worth a Changes entry).

dra27 reviewed (approved):
LGTM, thank you!
